app/src/main/java/com/arctouch/movies/home/view/fragment/MoviesFragment.java

```diff
@@ -0,0 +1,127 @@
+package com.arctouch.movies.home.view.fragment;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.arctouch.movies.R;
+import com.arctouch.movies.base.view.BaseFragment;
+import com.arctouch.movies.home.view.adapter.HomeMoviesAdapter;
+import com.arctouch.movies.movie.model.Movie;
+import com.arctouch.movies.movie.view.MovieDetailsActivity;
+
+import java.lang.annotation.Retention;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Created by marcelo.alves on 06/09/17.
+ */
+
+public class MoviesFragment extends BaseFragment implements HomeMoviesAdapter.OnMovieItemClickListener {
+
+    @Retention(SOURCE)
+    @IntDef({TYPE_MOVIES, TYPE_FAVORITES})
+    public @interface FragmentType {
+    }
+
+    public static final int TYPE_MOVIES = 0x01;
+    public static final int TYPE_FAVORITES = 0x02;
+
+    public static final String EXTRA_TYPE = "extra_type";
+
+    private View view;
+    private RecyclerView recyclerView;
+    private LinearLayoutManager layoutManager;
+    private HomeMoviesAdapter moviesAdapter;
+
+    private OnLoadMoreListener loadMoreListener;
+
+    private boolean isLoadingMore = false;
+
+    @FragmentType
+    private int type;
+
+    public static MoviesFragment newInstance(Bundle args, @Nullable OnLoadMoreListener listener) {
+        MoviesFragment fragment = new MoviesFragment();
+        fragment.setLoadMoreListener(listener);
```

Comment:
This is an issue: Fragments are designed to be destroyed and re-instantiated by the system, so their state can't be kept - that's why all "communication" with Fragments is done by argument bundles.
In practice, this means that you cannot guarantee that this listener will be there, so you need to find a different way of communicating back. I don't need to see code, but I'd like to have an idea of a possible solution :)

Reply:
Ok for this case We can have have two possible solution:

1 - Use EventBus to keep communication between Activity --> Fragment to notify any data change. 
2 - Do a refactoring and move the `search listener` and `fetch movies` calls to Fragment